zebrad/src/components/sync/downloads.rs

```diff
@@ -418,9 +436,17 @@ where
                 };
 
                 // Verify the block.
-                let rsp = verifier
+                let mut rsp = verifier
                     .map_err(|error| BlockDownloadVerifyError::VerifierServiceError { error })?
-                    .call(block);
+                    .call(block).boxed();
+
+                // Add a shorter timeout to workaround a known bug (#5125)
+                let short_timeout_max = (max_checkpoint_height + FINAL_CHECKPOINT_BLOCK_VERIFY_TIMEOUT_LIMIT).expect("checkpoint block height is in valid range");
+                if block_height >= max_checkpoint_height && block_height <= short_timeout_max {
+                    rsp = timeout(FINAL_CHECKPOINT_BLOCK_VERIFY_TIMEOUT, rsp)
+                        .map_err(|timeout| format!("initial fully verified block timed out: retrying: {:?}", timeout).into())
+                        .map(|nested_result| nested_result.and_then(convert::identity)).boxed();
+                }
```

Comment:
@arya2 or @upbqdn, once you're happy with the changes in commit 74b29be, you can resolve this comment, and the PR will start merging.

(I'm doing it this way because Arya can't approve PRs he has opened.)